cg/models/nf_analysis.py

```diff
@@ -44,22 +42,24 @@ def fastq_files_exist(cls, fastq_paths: List[str], values: dict) -> List[str]:
         return fastq_paths
 
 
-class NextflowDeliverables(BaseModel):
-    """Nextflow deliverables model
+class FileDeliverable(BaseModel):
+    """Specification for a general deliverables file."""
 
-    Attributes:
-        deliverables: dictionary containing format, path, path_index, step, tag and id keys.
-    """
+    id: str
+    format: str
+    path: FilePath
+    path_index: Optional[FilePath]
+    step: str
+    tag: str
+
+    @validator("path", "path_index")
+    def convert_path_to_string(cls, file_path):
+        if file_path and isinstance(file_path, Path):
+            return str(file_path)
+        return file_path
+
+
+class PipelineDeliverables(BaseModel):
+    """Specification for pipeline deliverables."""
 
-    deliverables: dict
-
-    @validator("deliverables")
-    def headers(cls, v: dict) -> None:
-        """Validate header format."""
-        if collections.Counter(list(v.keys())) != collections.Counter(DELIVER_FILE_HEADERS):
-            raise ValueError(
-                f"Headers are not matching the standard header format: {DELIVER_FILE_HEADERS}"
-            )
-        for key, value in v.items():
-            if not value and key != "path_index":
-                raise ValueError("An entry other than path_index is empty!")
+    files: List[FileDeliverable]
```

Comment:
this is no longer relevant since this has been moved, but I have added this suggestion for the related method `get_deliverables_template_content` but with `raise NonImplementedError` instead of pass